tests/foreman/ui_airgun/test_subscription.py

```diff
@@ -0,0 +1,171 @@
+"""Test class for Subscriptions/Manifests UI
+
+:Requirement: Subscription
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: UI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from robottelo import manifests
+from nailgun import entities
+
+from robottelo.datafactory import gen_string
+
+from robottelo.decorators import (
+    fixture,
+    tier1,
+    skip_if_not_set,
+    upgrade,
+)
+
+
+@fixture
+def organization():
+    return entities.Organization().create()
+
+
+@fixture
+def manifest_file():
+    with manifests.clone() as manifest:
+        with open(manifest.filename, 'wb') as handler:
+            handler.write(manifest.content.read())
+        return manifest.filename
+
+
+@skip_if_not_set('fake_manifest')
+@tier1
+@upgrade
+def test_positive_upload_and_delete(session, manifest_file, organization):
+    """Upload a manifest with minimal input parameters and delete it
+
+    :id: 58e549b0-1ba3-421d-8075-dcf65d07510b
+
+    :expectedresults: Manifest is uploaded and deleted successfully
+
+    :CaseImportance: Critical
+    """
+    with session:
+        session.organization.select(organization.name)
+        session.subscription.add_manifest(manifest_file)
+        assert session.subscription.has_manifest
+        session.subscription.delete_manifest()
+        assert not session.subscription.has_manifest
+
+
+@skip_if_not_set('fake_manifest')
+@tier1
+def test_positive_delete_confirmation(session, manifest_file, organization):
+    """Upload a manifest with minimal input parameters, press 'Delete'
+    button and check warning message on confirmation screen
+
+    :id: 16160ee9-f818-447d-b7ab-d04d396d50c5
+
+    :BZ: 1266827
+
+    :expectedresults: confirmation dialog contains informative message
+        which warns user about downsides and consequences of manifest
+        deletion
+
+    :CaseImportance: Critical
+    """
+    expected_message = [
+        'Are you sure you want to delete the manifest?',
+        'Note: Deleting a subscription manifest is STRONGLY discouraged. '
+        'Deleting a manifest will:',
+        'Delete all subscriptions that are attached to running hosts.',
+        'Delete all subscriptions attached to activation keys.',
+        'Disable Red Hat Insights',
+        'Require you to upload the subscription-manifest and re-attach '
+        'subscriptions to hosts and activation keys.',
+        'This action should only be taken in extreme circumstances or for '
+        'debugging purposes.',
+    ]
+    with session:
+        session.organization.select(organization.name)
+        session.subscription.add_manifest(manifest_file)
```

Comment:
think we do not need to add manifest via UI in this test, we can use API
we can use upload_manifest_locked function from manifests modules, in other case we have to decorate this function with run_in_one_thread.